src/rules/no-duplicate-definitions.js

```diff
@@ -85,11 +85,11 @@ export default {
 			),
 		);
 
-		/** @type {Set<string>} */
-		const definitions = new Set();
+		/** @type {Map<string, { line: number, label: string }>} */
+		const definitions = new Map();
 
-		/** @type {Set<string>} */
-		const footnoteDefinitions = new Set();
+		/** @type {Map<string, { line: number, label: string }>} */
```

Comment:
```suggestion
		/** @type {Map<string, FootnoteDefinition>} */
```

Maybe the final code would look like this:

<img width="686" height="457" alt="image" src="https://github.com/user-attachments/assets/aeab61ee-2152-4b42-8b3f-865fbe73d615" />